$OpenBSD: patch-lib_puppet_provider_service_openbsd_rb,v 1.12 2014/05/15 20:17:16 jasper Exp $

* https://github.com/puppetlabs/puppet/pull/2557
- Do not unconditionally alphabetically sort the list of pkg_scripts
  as order matters. Explicit ordering will need to be forced through
  resources dependencies.
- Do not add _flags="" for pkg_scripts if no :flags were set, as it
  may override the default flags from inside the script.
- Use a faster way of determining if something is in base, instead of checking
  something is from packages with pkg_info.

--- lib/puppet/provider/service/openbsd.rb.orig	Thu May 15 21:04:49 2014
+++ lib/puppet/provider/service/openbsd.rb	Thu May 15 21:04:52 2014
@@ -172,7 +172,9 @@ Puppet::Type.type(:service).provide :openbsd, :parent 
     end
 
     if flags.nil?
-      append = resource[:name] + '_flags=""'
+      if in_base?
+	append = resource[:name] + '_flags=""'
+     end
     else
       append = resource[:name] + '_flags="' + flags + '"'
     end
@@ -208,7 +210,7 @@ Puppet::Type.type(:service).provide :openbsd, :parent 
   # return the array with the current resource added
   # @api private
   def pkg_scripts_append
-    [pkg_scripts(), resource[:name]].flatten.sort.uniq
+    [pkg_scripts(), resource[:name]].flatten.uniq
   end
 
   # return the array without the current resource
@@ -243,12 +245,11 @@ Puppet::Type.type(:service).provide :openbsd, :parent 
     content
   end
 
-  # Determine if the rc script is included in base, or if it exists as a result
-  # of a package installation.
+  # Determine if the rc script is included in base
   # @api private
   def in_base?
-    system("/usr/sbin/pkg_info -qE /etc/rc.d/#{self.name}> /dev/null")
-    $?.exitstatus == 1
+    script = File.readlines(self.class.rcconf).find {|s| s =~ /^#{rcvar_name}/ }
+    !script.nil?
   end
 
   # @api private
